javaslang/src/main/java/javaslang/collection/List.java

```diff
@@ -812,11 +817,15 @@ default int lastIndexOf(T element, int end) {
     @Override
     default Tuple2<List<T>, List<T>> partition(Predicate<? super T> predicate) {
         Objects.requireNonNull(predicate, "predicate is null");
-        final java.util.List<T> left = new ArrayList<>(), right = new ArrayList<>();
+        Queue<T> left = Queue.empty(), right = Queue.empty();
         for (T t : this) {
-            (predicate.test(t) ? left : right).add(t);
+            if (predicate.test(t)) {
+                left = left.append(t);
+            } else {
+                right = right.append(t);
+            }
         }
-        return Tuple.of(List.ofAll(left), List.ofAll(right));
+        return Tuple.of(left.toList(), right.toList());
     }
```

Comment:
We should save even more instances by directly prepending to Lists instead of appending to Queues:

```java
List<T> left = empty(), right = empty();
for (T t : this) {
    if (predicate.test(t)) {
        left = left.prepend(t);
    } else {
        right = right.prepend(t);
    }
}
return Tuple.of(left.reverse(), right.reverse());
```